File: D1/SimpleFactory/Archer.py
class Archer:
    def get_type(self):
        pass

class Warrior(Archer):
    def get_type(self):
        print("This is Warrior")
        return type(self).__name__

class Adventurer(Archer):
    def get_type(self):
        print("This is Adventurer")
        return type(self).__name__

File: DESIGN-CSharp/SimpleFactory/Archer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SimpleFactory
{
    class Archer : Adventurer
    {
        public override string getType()
        {
            Console.WriteLine("this is Archar ");
            return this.GetType().Name;
        }
    }
}
